### DISCUSSION TITLE
Wax removal-like effect

### DISCUSSION THREAD

**Discussion by edazpotato** [FabricMC/fabric-api#1643] [Mod Dev Support]
How would I go about creating a block that when it is right-clicked by a player with an axe, the block is replaced with a different block? (Similar to the way one removes wax from copper blocks)

**Comment by warjort** [comment]
Doing it exactly the way minecraft does it is not straightforward.
See the discussion here about HoneycombItem:
https://github.com/FabricMC/fabric/pull/1632

But you can still do it yourself by implementing the Block.onUse() method for your block.

The usual pattern is:
````
   @Override
   public ActionResult onUse(BlockState state, World world, BlockPos pos, PlayerEntity player, Hand hand, BlockHitResult hit) {
      // The client says it is ok, so it will pass to the server (also makes the player's arm swing)
      if (world.isClient) {
        return ActionResult.SUCCESS;
      }
      // Server code here where you can change the block state if required
   }
````

You get the itemStack to test if it is an axe using ````player.getItemInHand(hand)````

**Comment by warjort** [comment]
If you want to do the same thing for something that is not your block, you will need to register this callback:
https://github.com/FabricMC/fabric/blob/1.17/fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/player/UseBlockCallback.java
which will give you an event for EVERY block use, so be careful.
